spec/services/change_offer_spec.rb

```diff
@@ -0,0 +1,42 @@
+require 'rails_helper'
+
+RSpec.describe ChangeOffer do
+  include CourseOptionHelpers
+
+  let(:application_choice) { create(:application_choice) }
+  let(:course_option) { course_option_for_provider(provider: application_choice.course_option.provider) }
+  let(:provider_user) { create(:provider_user, providers: [build(:provider)]) }
+  let(:conditions) { [Faker::Lorem.sentence] }
+
+  let(:change_offer) do
+    ChangeOffer.new(actor: provider_user,
+                    application_choice: application_choice,
+                    course_option: course_option,
+                    conditions: conditions)
+  end
+
+  describe '#save!' do
+    it 'calls save on the ChangeAnOffer service with no errors' do
+      change_an_offer = instance_double(ChangeAnOffer)
+      allow(change_an_offer).to receive(:save).and_return(true)
+      allow(ChangeAnOffer).to receive(:new) { change_an_offer }
+
+      expect { change_offer.save! }.not_to raise_error
+
+      expect(ChangeAnOffer).to have_received(:new).with(actor: provider_user,
+                                                        application_choice: application_choice,
+                                                        course_option: course_option,
+                                                        offer_conditions: conditions)
+
+      expect(change_an_offer).to have_received(:save)
+    end
+
+    it 'raises an error when the save is unsuccessful' do
+      change_an_offer = instance_double(ChangeAnOffer)
+      allow(change_an_offer).to receive(:save).and_return(false)
+      allow(ChangeAnOffer).to receive(:new) { change_an_offer }
+
+      expect { change_offer.save! }.to raise_error('Unable to complete save on change_an_offer')
+    end
+  end
+end
```

Comment:
This looks ok, however I think that similar to the make offer spec - prior to the changes included in this PR - it should ensure that the correct set of services is called (e.g. it should set some expectations for the calls to SetDeclineByDefault, CandidateMailer and StateChangeNotifier). This way if any further changes take place in ChangeAnOffer for any reason we will be aware and none of the expected ChangeOffer functionality will change.